### ISSUE TITLE
[Feature Request] Swagger UI integration/example

### ISSUE THREAD

**Issue by netthier** [tamasfe/aide#65]
Since its [most recent release](https://github.com/swagger-api/swagger-ui/releases/tag/v5.0.0) Swagger UI supports OpenAPI 3.1 specs, meaning it can be used to render aide-generated docs.
As it offers a few useful features over Redoc, such as a convenient "Try it out" button, it would be interesting to have support for this similar to `aide::redoc` or at least docs in the form of an example using it.
However, the build process seems a bit more involved. While there is a static dist version similar to how Redoc works, it's recommended to use the npm package in conjunction with a tool like webpack instead to save space, [as described here](https://github.com/swagger-api/swagger-ui#general).
Not sure what the best way to go forward is because of that.

**Comment by kaujea** [comment]
Hey!
I took a look at this issue and I believe that an integration in aide would fall under the category "serve Swagger UI in a server-side project", for which they recommend the dist bundle (similar to the `aide::redoc`).
I created a [proof of concept](https://github.com/kaujea/aide/tree/feature/swagger-ui), based what is done in utoipa. I'd be delighted to open a PR here! I am new to open source so I might require some support to get everything right.

**Comment by Wicpar** [comment]
Thank you @kaujea !
There is only an issue with your method: it breaks standalone artifacts since it relies on a folder existing at runtime. I think it would be far simpler as well to include_str each file from the res folder so that it is compile inside of the executable, just like redoc is done right now.
If you do that i'll accept the PR

**Comment by kaujea** [comment]
@Wicpar 
Fair point. I tried to address the issue, but it got a bit more verbose than I hoped for (esp. I had to resort to a macro).
Let me know if you see a better way to achieve this result.

**Comment by LehMaxence** [comment]
I am not a lawyer but:

- of course, you don't have to change the license of aide's IP (MIT Or Apache-2.0)
- including swagger-ui inside the crate, i.e. the distributed file, makes the Apache 2.0 always applicable, and its NOTICE file must be part of the distribution, so the license of the crate becomes (MIT OR Apache-2.0) AND Apache-2.0

You can check [unicode-ident](https://github.com/dtolnay/unicode-ident/) as reference of a crate including a third-party work.

**Comment by Wicpar** [comment]
If we include the source of course they will be attached to the license, either within the source or the folder, and it should be mentionned in the Readme, but the crates license remains the same. If swagger has issues with that they may get in touch with us.

**Comment by Wicpar** [comment]
I'm going to integrate the feature using https://crates.io/crates/swagger-ui, it's a lot more compliant and maintanable as it pulls directly from NPM.

**Comment by daniel7grant** [comment]
Hey @Wicpar, did you eventually come up with something regarding this? I would love to have a SwaggerUI integration in `aide`. If you like, I could put together a PR, based on the other integrations (Redoc, Scalar).

**Comment by Swivelgames** [comment]
I was actually super hopeful when I found `aide` that it was able to serve a Swagger UI.

Any progress on this @Wicpar? Not sure how helpful I'll be, but is there any assistance I can provide?

**Comment by devmaxde** [comment]
I made a simple implementation without the npm crate @Wicpar mentioned. Maybe worth a look

**Comment by JakkuSakura** [comment]
@devmaxde thanks for your effort. the PR is now merged
